### PR TITLE
Release 2.0.7

### PR DESCRIPTION
**Pull request by skion**
This cherry picks the PRs mentioned in the changelog into the 2.x branch.

Following [semantic versioning](https://semver.org/), not included are:

* OpenID Connect jwt (#488) (not sure if backward compatible, for 3.0.0)
* Fix `client_id` in web request body (#505) (not sure if backward compatible, for 3.0.0)
* Add support for HMAC-SHA256 (builds on #388) (#498) (new feature, for 2.1.0)

### CODE REVIEW

**Comment by JonathanHuot** [comment]
LGTM

**Comment by wiliamsouza** [comment]
:+1:

**Comment by duaneking** [comment]
Have we gotten sign off this introduces no compatability issues with our largest downstream users?
Maybe we should tag them and ask them to put this through its paces before official release?

@evonove
@lepture

**Comment by skion** [comment]
Yes, I've run the test suits for `requests-oauthlib`, `django-oauth-toolkit` and `flask-oauthlib` without problems on this version...

**Comment by JonathanHuot** [comment]
Any blocking changes ? 

I think this PR can be accepted safely (into 2.x branch) and also oauthlib/oauthlib#521 oauthlib/oauthlib#523 oauthlib/oauthlib#520 oauthlib/oauthlib#522 oauthlib/oauthlib#518 into master. Those are around build/docs tooling, they might be needed before tagging the branch 2.x into 2.0.7.

Will be great if someone could cope with these PR, @thedrow ? any other with write permissions ?

**Comment by thedrow** [comment]
@JonathanHuot The PRs mentioned here introduce new features so we'll need to declare the version as 2.1.0.
We follow SemVer as our versioning scheme.

**Comment by skion** [comment]
Hm, which one(s) specifically @thedrow?

**Comment by skion** [comment]
[Here](https://github.com/oauthlib/oauthlib/milestone/1)'s a milestone for this release...

**Comment by thedrow** [comment]
In that case, everything looks good. I thought you guys wanted to backport the PRs in the OP to 2.0.7.

**Comment by skion** [comment]
I've picked the recent non-breaking PRs, and am only waiting to merge #518 to be tested.

We also may have something to fix in PyPi still: #526

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 * Added bottle-oauthlib to docs. (#509)
-* Updated docs for organization change. (#515)
 * Update repository location in Travis. (#514)
+* Updated docs for organization change. (#515)
 * Replace G+ with Gitter. (#517)
+* Update requirements. (#518)
+* Add shields for Python versions, license and RTD. (#520)
+* Fix ReadTheDocs build (#521).
+* Fixed "make" command to test upstream with local oauthlib. (#522)
+* Replace IRC notificatgion with Gitter Hook. (#523)
```
Comment (thedrow) [inline CHANGELOG.rst]: There's a typo here. notificatgion -> notification.

Comment (skion) [reply CHANGELOG.rst]: Fixed

**Pull request merged**